filebeat/prospector/factory.go

```diff
@@ -0,0 +1,38 @@
+package prospector
+
+import (
+	"github.com/elastic/beats/filebeat/registrar"
+	"github.com/elastic/beats/libbeat/cfgfile"
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type Factory struct {
+	outlet    Outlet
+	registrar *registrar.Registrar
+}
+
+func NewFactory(outlet Outlet, registrar *registrar.Registrar) *Factory {
+
+	return &Factory{
+		outlet:    outlet,
+		registrar: registrar,
+	}
+}
+
+func (r *Factory) Create(c *common.Config) (cfgfile.Runner, uint64, error) {
```

Comment:
with `Runner` implementing `ID()`, we need to return an ID here? Or put alternatively: do we need `Runner` to implement `ID()` ?